src/Optimizer/Transformer/AmpStoryCssOptimizer.php

```diff
@@ -0,0 +1,244 @@
+            if (! $hasAmpStoryDvhPolyfillScript) {
+                $this->appendAmpStoryDvhPolyfillScript($document);
+            }
+        }
+
+        $this->supportsLandscapeSSR($document);
+        $this->aspectRatioSSR($document);
+    }
+
+    /**
+     * Check whether the element is an AMP Story element.
+     *
+     * @param Element $element Element to check.
+     * @return bool Whether the given element is an AMP story.
+     */
+    private function isAmpStoryScript(Element $element)
+    {
+        return ($element->tagName === Tag::SCRIPT)
+            && ($element->getAttribute(Attribute::CUSTOM_ELEMENT) === Extension::STORY);
+    }
+
+    /**
+     * Check whether the element is a script[amp-story-dvh-polyfill] element.
+     *
+     * @param Element $element Element to check.
+     * @return bool Whether the element is a script[amp-story-dvh-polyfill] element.
+     */
+    private function isAmpStoryDvhPolyfillScript(Element $element)
+    {
+        return $element->tagName === Tag::SCRIPT && $element->hasAttribute(Attribute::AMP_STORY_DVH_POLLYFILL);
+    }
+
+    /**
+     * Check whether the element is a style[amp-custom] element.
+     *
+     * @param Element $element Element to check.
+     * @return bool Whether the element is a style[amp-custom] element.
+     */
+    private function isStyleAmpCustom(Element $element)
+    {
+        return $element->tagName === Tag::STYLE && $element->hasAttribute(Attribute::AMP_CUSTOM);
+    }
+
+    /**
+     * Insert a link element with amp-story css source.
+     *
+     * @param Document $document Document to append the link.
+     */
+    private function appendAmpStoryCssLink(Document $document)
+    {
+        $ampStoryCssLink = $document->createElementWithAttributes(Tag::LINK, [
+            Attribute::REL           => Attribute::REL_STYLESHEET,
+            Attribute::AMP_EXTENSION => Extension::STORY,
+            Attribute::HREF          => Amp::CACHE_HOST . '/v0/amp-story-1.0.css',
+        ]);
+
+        $document->head->appendChild($ampStoryCssLink);
+    }
+
+    /**
+     * Replace viewport units in custom css with related css variables.
+     *
+     * @param Element $style The style element to modify.
+     */
+    private function modifyAmpCustomCSS(Element $style)
+    {
+        $style->nodeValue = preg_replace(
+            '/(-?[\d.]+)v(w|h|min|max)/',
+            'calc($1 * var(--story-page-v$2))',
+            $style->nodeValue
+        );
+    }
+
+    /**
+     * Append an inline script tag for the dvh polyfill
+     *
+     * @param Document $document The document in which we need to append the script tag.
+     * @return void
+     */
+    private function appendAmpStoryDvhPolyfillScript(Document $document)
+    {
+        $ampStoryDvhPolyfillScript = $document->createElementWithAttributes(Tag::SCRIPT, [
+            Attribute::AMP_STORY_DVH_POLLYFILL => '',
+        ], self::AMP_STORY_DVH_POLYFILL_CONTENT);
```

Comment:
Make formatting more consistent with rest of code:
```suggestion
        $ampStoryDvhPolyfillScript = $document->createElementWithAttributes(
            Tag::SCRIPT,
            [
                Attribute::AMP_STORY_DVH_POLLYFILL => '',
            ],
            self::AMP_STORY_DVH_POLYFILL_CONTENT
        );
```